pallets/parachain-staking/src/lib.rs

```diff
@@ -507,6 +507,27 @@ pub mod pallet {
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		// This upgrade fixes a bug that may have led to an incorrect `Total` (total staked)
+		fn on_runtime_upgrade() -> Weight {
+			let old_total = Total::<T>::get();
+			let mut new_total: BalanceOf<T> = 0u32.into();
+
+			for collator_state in CollatorState::<T>::iter_values() {
+				new_total += collator_state.total;
+			}
+
+			Total::<T>::put(new_total);
+
+			log::trace!(
+				target: "staking",
+				"Finished migrating storage.\nOld Total : {:?}\nNew Total : {:?}",
+				old_total,
+				new_total,
+			);
+
+			10_000
```

Comment:
We should put something close to max block weight here just to be safer until we get better at upgrades.

Reply:
```suggestion
			300_000_000_000 // Three fifths of the max block weight
```

This is just less than 3/4 (which is the normal ratio) of max block weight. Should be plenty of time to migrate alphanet's collators.